scipy/linalg/decomp_svd.py

```diff
@@ -279,6 +286,98 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def higher_order_svd(
+    a: npt.ArrayLike, *, full_tensor: bool = True, check_finite: bool = True
+) -> Tuple[List[numpy.ndarray], numpy.ndarray]:
+    """Higher-order SVD (HOSVD)
+
+    Factorizes the M-D tensor `a` into a list of M matrices ``U_k`` containing
+    the left singular vectors of the unfolded tensor with respect to the
+    ``k``\\ th axis and a core tensor ``S`` such that
+    :math:`a = (U_1, U_2, \\dots, U_M) \\cdot S`.
+
+    Parameters
+    ----------
+    a : array_like
+        Tensor to decompose
+    full_tensor : bool, optional
+        If True (default), ``S`` is of the same shape as the input tensor `a`
+        and all ``U_k`` are of the shape ``(n, n)`` with ``n`` corresponding to
+        the ``k``\\ th axis of ``S``.
+        If False, the shapes of the ``U_k`` are ``(n, m)`` with ``n``
+        corresponding to the size of the ``k``\\ th axis of the input `a` and
+        ``m`` to the maybe reduced size of the ``k``\\ th axis of the core
+        tensor.
+    check_finite : bool, optional
+        Whether to check that the input matrix contains only finite numbers.
+        Disabling may give a performance gain, but may result in problems
+        (crashes, non-termination) if the inputs do contain infinities or NaNs.
+
+    Returns
+    -------
+    U : list of ndarray
+        List of the unitary matrices ``U_k`` with left singular vectors of the
+        tensor unfolding corresponding to the `k`\\ th axis.
+        Of shape ``(M, M)`` or ``(M, K)``, depending on `full_tensor`.
+    s : ndarray
```

Comment:
Should be consistent with the doc so S I guess.
```suggestion
    S : ndarray
```

Reply:
Changed.